pkg/controller/sharedinfo/shared_info.go

```diff
@@ -4,22 +4,77 @@
 package sharedinfo
 
 import (
+	"context"
+	"fmt"
+	"strings"
+
+	"gopkg.in/ini.v1"
+
 	configv1 "github.com/openshift/api/config/v1"
+	operatortypes "github.com/vmware/nsx-container-plugin-operator/pkg/types"
+	operatorv1 "github.com/vmware/nsx-container-plugin-operator/pkg/apis/operator/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
+var log = logf.Log.WithName("shared_info")
+
 type SharedInfo struct {
+	AdaptorName           string
+	AddNodeTag            bool
+	NetworkConfig         *configv1.Network
+	OperatorConfigMap     *corev1.ConfigMap
+	OperatorNsxSecret     *corev1.Secret
+
 	NsxNodeAgentDsSpec    *unstructured.Unstructured
 	NsxNcpBootstrapDsSpec *unstructured.Unstructured
-
-	NsxNcpDeploymentSpec *unstructured.Unstructured
-
-	NetworkConfig     *configv1.Network
-	OperatorConfigMap *corev1.ConfigMap
-	OperatorNsxSecret *corev1.Secret
+	NsxNcpDeploymentSpec  *unstructured.Unstructured
 }
 
-func New() *SharedInfo {
-	return &SharedInfo{}
+func New(mgr manager.Manager, operatorNamespace string) (*SharedInfo, error) {
+	reader := mgr.GetAPIReader()
+	configmap := &corev1.ConfigMap{}
+	watchedNamespace := operatorNamespace
+	if watchedNamespace == "" {
+		log.Info(fmt.Sprintf("SharedInfo can only check a single namespace, defaulting to: %s",
+			operatortypes.OperatorNamespace))
```

Comment:
miss a whitespace

Reply:
talked with Danting offline. won't modify it because the indent is tab, not whilespaces.